app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt

```diff
@@ -93,6 +97,12 @@ class EventsFragment : Fragment() {
             showNoInternetScreen(isNetworkConnected())
         }
 
+        rootView.swiperefresh.setColorSchemeColors(Color.BLUE)
+        rootView.swiperefresh.setOnRefreshListener({
+            eventsViewModel.loadLocationEvents(refreshLocationValue)
+            rootView.swiperefresh.isRefreshing = false
+        })
```

Comment:
Memory leak

Reply:
Rotate and verify that it's working. Also, refreshing should stop after actual refreshing is complete. Not just by default

Reply:
If I rotate and then refresh it reloads the events according to the previous location that is stored in the sharedPref.
I am a little confused why this is happening because it loads correctly if I don't rotate the screen

and if I don't change the `isRefreshing` value to false the progress bar keeps rotating forever even though the refresh has completed.

Reply:
First problem is not related to the swipe refresh view, and obviously if you don't set to false, it'll keep rotating, read my previous comment